src/RealPath.php

```diff
@@ -24,74 +23,28 @@
 
 /**
  * @psalm-type Options = array{
- *     exists: bool,
- *     ...
+ *     exists?: bool,
  * }
- * @template TOptions of Options
- * @extends AbstractFilter<TOptions>
+ * @implements FilterInterface<string|array<array-key, string|mixed>>
```

Comment:
```suggestion
 * @implements FilterInterface<string>
```

The successfully filtered value is a string - therefore the return type of `filter()` will be `mixed|string` - we might be able to narrow this to `non-empty-string` further if its not a massive pain in the ass.